c7n/resources/cloudtrail.py

```diff
@@ -31,3 +36,107 @@ class resource_type(object):
         name = 'Name'
         dimension = None
         config_type = "AWS::CloudTrail::Trail"
+
+    filter_registry = filters
+
+
+@filters.register('monitored-metric')
+class MonitoredCloudtrailMetric(ValueFilter):
+    """Finds cloudtrails with logging and a metric filter. Is a subclass of ValueFilter,
+    filtering the metric filter objects. Optionally, verifies an alarm exists (true by default),
+    and for said alarm, there is atleast one SNS subscription (again, true by default).
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: cloudtrail-trail-with-login-attempts
+                resource: cloudtrail
+                region: us-east-1
+                filters:
+                  - type: monitored-metric
+                    alarm: true
+                    topic-subscription: false
+                    filter: '$.eventName = DeleteTrail'
+    """
+
+    schema = type_schema('monitored-metric', rinherit=ValueFilter.schema, **{
+        'topic-subscription': {'type': 'boolean'},
+        'alarm': {'type': 'boolean'}
+    })
+
+    permissions = ('logs:DescribeMetricFilters', 'cloudwatch:DescribeAlarms',
+        'sns:ListSubscriptionsByTopic')
+
+    def _filterTopicArnsToSubscribed(self, session, topicArns):
+        sns = session.client('sns')
+
+        def arnHasSubscriptions(arn):
+            subscriptions = sns.list_subscriptions_by_topic(TopicArn=arn)['Subscriptions']
+            return any(subscriptions)
+        return filter(arnHasSubscriptions, topicArns)
+
+    def _allAlarms(self, session):
+        # TODO: We should cache this better.
+        cloudwatch = session.client('cloudwatch')
```

Comment:
I think using the build in resource managers for alarms, filters and possibly the topic stuff would be useful - that way we can intelligently do the join in python and keep the data cached. Otherwise it's many requests per trail.